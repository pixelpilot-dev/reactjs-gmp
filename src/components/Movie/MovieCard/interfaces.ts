import { IMovieProps } from '../../../types/IMovieProps';

export interface IMovieCardProps {
  movie: IMovieProps;
  className?: string;
  onEdit: (id: string) => void;
  onDelete: (id: string) => void;
}
