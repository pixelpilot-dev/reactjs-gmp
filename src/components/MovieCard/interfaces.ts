import { IMovieProps } from '../../types/IMovieProps';

export interface IMovieCardProps {
  movie: IMovieProps;
  className?: string;
}
