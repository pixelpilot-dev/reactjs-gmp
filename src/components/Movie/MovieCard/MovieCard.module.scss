.card {
  position: relative;

  .preview {
    display: block;
    width: 100%;
    margin-bottom: 20px;
    overflow: hidden;

    img {
      transition: transform 0.3s ease-in;
    }

    @include for-hover {
      &:hover img {
        transform: scale(1.02);
      }
    }
  }

  .info {
    display: flex;
    justify-content: space-between;
  }

  .title {
    display: block;
    color: $white7;
    margin-bottom: 8px;
    margin-right: 5px;
    transition: color 0.3s linear;

    @include for-hover {
      &:hover {
        color: $primary;
      }
    }
  }

  .genre {
    @extend %p-hint;

    color: $white3;

    &::after {
      content: ', ';
    }

    &:last-child::after {
      display: none;
    }
  }

  .year {
    @extend %p-hint;

    display: inline-block;
    border: 1px solid $border-grey;
    border-radius: 4px;
    text-align: center;
    color: $white7;
    width: 66px;
    height: 26px;
    line-height: 26px;
  }

  .menuContext {
    position: absolute;
    top: 14px;
    right: 14px;
    z-index: 1;
    width: 190px;
    opacity: 0;
    visibility: hidden;
    transition: opacity 0.2s linear, visibility 0.2s linear;
  }

  .actionBtn {
    @include resetbtn;
    @extend %p-md;
    background-color: transparent;
    color: $white;
    padding: 7px 20px;
    text-align: left;
    transition: background-color 0.3s linear;

    @include for-hover {
      &:hover {
        background-color: $primary;
      }
    }
  }

  @include for-hover {
    &:hover .menuContext {
      opacity: 1;
      visibility: visible;
    }
  }
}
