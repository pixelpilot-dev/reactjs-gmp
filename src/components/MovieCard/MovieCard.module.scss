.card {
  .preview {
    display: block;
    width: 100%;
    margin-bottom: 20px;
    overflow: hidden;

    img {
      transition: transform 0.3s ease-in;
    }

    &:hover img {
      transform: scale(1.02);
    }
  }

  .info {
    display: flex;
    justify-content: space-between;
  }

  .title {
    display: block;
    color: $white7;
    margin-bottom: 8px;
    margin-right: 5px;
    transition: color 0.3s linear;

    &:hover {
      color: $primary;
    }
  }

  .genre {
    @extend %p-hint;
    color: $white3;

    &::after {
      content: ', ';
    }

    &:last-child::after {
      display: none;
    }
  }

  .year {
    @extend %p-hint;
    display: inline-block;
    border: 1px solid $border-grey;
    border-radius: 4px;
    text-align: center;
    color: $white7;
    width: 66px;
    height: 26px;
    line-height: 26px;
  }
}
