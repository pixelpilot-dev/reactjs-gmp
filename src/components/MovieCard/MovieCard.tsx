import * as React from 'react';
import cn from 'classnames';
import genresList from '../../data/genre.json';
import { IMovieCardProps } from './IMovieCardProps';
import { getYear } from '../../utils/getYear';

import styles from './MovieCard.module.scss';

export const MovieCard: React.FC<IMovieCardProps> = ({ movie, className }) => {
  const { url, title, preview, genre, releaseDate } = movie;

  const renderGenres = genres => {
    return genres.map(item => (
      <span key={item} className={styles.genre}>
        {genresList[item]}
      </span>
    ));
  };

  return (
    <div className={cn(styles.card, className)}>
      <a href={url} className={styles.preview}>
        <img src={preview} alt={title} />
      </a>

      <div className={styles.info}>
        <a href={url} className={styles.title}>
          {title}
        </a>
        <span className={styles.year}>{getYear(releaseDate)}</span>
      </div>
      <div className={styles.genres}>{renderGenres(genre)}</div>
    </div>
  );
};
