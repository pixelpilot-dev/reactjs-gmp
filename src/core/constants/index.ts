import { range } from 'lodash';
import { getYear } from '../utils/date';

export const MONTH = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December',
];

export const YEARS = range(1700, getYear(new Date()) + 1, 1);

export const GENRES = [
  'Action',
  'Adventure',
  'Drama',
  'Biography',
  'Music',
  'Oscar winning Movie',
  'Documentary',
  'Comedy',
  'Horror',
  'Crime',
  'Romance',
  'Animation',
  'Family',
  'Fantasy',
  'Science Fiction',
  'History',
  'Thriller',
  'Mystery',
  'Film-Noir',
  'War',
  'Western',
  'Musical',
  'Sport',
];

export const ERROR_CAPTION_FOR_FORMS = {
  REQUIRED: 'Required',
  IS_EMAIL: 'Email is invalid',
  IS_NUMBER: 'Should only contain numbers',
  IS_MIN: 'Must be greader {0}',
  IS_MAX: 'Must be less than or equal to {0}',
  IS_URL: 'Must be url',
};

export const NOTIFICATION_TYPES = {
  SUCCESS: 'success',
  ERROR: 'error',
  WARNING: 'warning',
};
