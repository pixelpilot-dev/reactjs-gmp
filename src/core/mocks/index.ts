export const MOVIE = {
  id: 406997,
  title: 'Wonder',
  tagline: 'Are you ready to meet Auggie Pullman?',
  vote_average: 8.2,
  vote_count: 1620,
  release_date: '2017-11-13',
  poster_path: 'https://image.tmdb.org/t/p/w500/ouYgAatYH4JzIThj6FI3UYf31RI.jpg',
  overview: 'The story of August Pullman – a boy with facial differences – who enters fifth grade, attending a mainstream elementary school for the first time.',
  budget: 20000000,
  revenue: 0,
  genres: ['Drama', 'Family'],
  runtime: 113
};

export const MOVIES = [{"id":412302,"title":"Gemini","tagline":"","vote_average":10,"vote_count":3,"release_date":"2018-03-30","poster_path":"https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg","overview":"A heinous crime tests the complex relationship between a tenacious personal assistant and her Hollywood starlet boss. As the assistant travels across Los Angeles to unravel the mystery, she must stay one step ahead of a determined policeman and confront her own understanding of friendship, truth and celebrity.","budget":0,"revenue":0,"genres":["Mystery","Thriller"],"runtime":92},{"id":19404,"title":"Dilwale Dulhania Le Jayenge","tagline":"Come... Fall in Love","vote_average":9.2,"vote_count":1297,"release_date":"1995-10-20","poster_path":"https://image.tmdb.org/t/p/w500/uC6TTUhPpQCmgldGyYveKRAu8JN.jpg","overview":"Raj is a rich, carefree, happy-go-lucky second generation NRI. Simran is the daughter of Chaudhary Baldev Singh, who in spite of being an NRI is very strict about adherence to Indian values. Simran has left for India to be married to her childhood fiancé. Raj leaves for India with a mission at his hands, to claim his lady love under the noses of her whole family. Thus begins a saga.","budget":13200000,"revenue":100000000,"genres":["Comedy","Drama","Romance"],"runtime":190},{"id":425895,"title":"Tisay","tagline":"","vote_average":9,"vote_count":1,"release_date":"2016-11-14","poster_path":"https://image.tmdb.org/t/p/w500/lt03HzS8tfDg5ipUQcLqIRNs0sC.jpg","overview":"In the world of semi-pro basketball, a beautiful bookie recruits a naïve but promising player to the underworld of game-fixing.","budget":0,"revenue":0,"genres":["Drama","Romance"],"runtime":100},{"id":511679,"title":"Héctor El Father: Conocerás la verdad","tagline":"","vote_average":8.8,"vote_count":5,"release_date":"2018-03-22","poster_path":"https://image.tmdb.org/t/p/w500/jwJoURyfm4XxtnYRtIOS2pYR9Np.jpg","overview":"","budget":0,"revenue":0,"genres":["Drama"],"runtime":90},{"id":278,"title":"The Shawshank Redemption","tagline":"Fear can hold you prisoner. Hope can set you free.","vote_average":8.6,"vote_count":9868,"release_date":"1994-09-23","poster_path":"https://image.tmdb.org/t/p/w500/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg","overview":"Framed in the 1940s for the double murder of his wife and her lover, upstanding banker Andy Dufresne begins a new life at the Shawshank prison, where he puts his accounting skills to work for an amoral warden. During his long stretch in prison, Dufresne comes to be admired by the other inmates -- including an older prisoner named Red -- for his integrity and unquenchable sense of hope.","budget":25000000,"revenue":28341469,"genres":["Drama","Crime"],"runtime":142},{"id":238,"title":"The Godfather","tagline":"An offer you can't refuse.","vote_average":8.5,"vote_count":7437,"release_date":"1972-03-14","poster_path":"https://image.tmdb.org/t/p/w500/rPdtLWNsZmAtoZl9PK7S2wE3qiS.jpg","overview":"Spanning the years 1945 to 1955, a chronicle of the fictional Italian-American Corleone crime family. When organized crime family patriarch, Vito Corleone barely survives an attempt on his life, his youngest son, Michael steps in to take care of the would-be killers, launching a campaign of bloody revenge.","budget":6000000,"revenue":245066411,"genres":["Drama","Crime"],"runtime":175},{"id":372058,"title":"Your Name.","tagline":"","vote_average":8.5,"vote_count":2129,"release_date":"2016-08-26","poster_path":"https://image.tmdb.org/t/p/w500/xq1Ugd62d23K2knRUx6xxuALTZB.jpg","overview":"High schoolers Mitsuha and Taki are complete strangers living separate lives. But one night, they suddenly switch places. Mitsuha wakes up in Taki’s body, and he in hers. This bizarre occurrence continues to happen randomly, and the two must adjust their lives around each other.","budget":0,"revenue":355398372,"genres":["Romance","Animation","Drama"],"runtime":106},{"id":418827,"title":"Sister's Younger Husband","tagline":"A sexy brother-in-law approached her? And he’s a younger man?","vote_average":8.5,"vote_count":2,"release_date":"2016-10-11","poster_path":"https://image.tmdb.org/t/p/w500/wKCkyb7dyDTbZn7qy6zh8oqtFrT.jpg","overview":"A sexy brother-in-law approached her? And he’s a younger man?  Mi-yeon, who was adopted into her family when she was little, is living with her older sister, Si-yeon and Si-yeon’s husband. Mi-yeon’s desire towards her brother-in-law grows more and more every night when she gets to watch the affectionate interaction between her older sister and the brother-in-law. One day, Mi-yeon seduces her brother-in-law and the two end up crossing the line when they should not have. As the situation goes on, Mi-yeon finds out her older sister is also having affairs outside the marriage. Mi-yeon tries to clear up the relationship with her brother-in-law out of guilty feeling. However, she hears something shocking from the brother-in-law.","budget":0,"revenue":0,"genres":["Comedy","Drama","Romance"],"runtime":75},{"id":129,"title":"Spirited Away","tagline":"","vote_average":8.4,"vote_count":5095,"release_date":"2001-07-20","poster_path":"https://image.tmdb.org/t/p/w500/dL11DBPcRhWWnJcFXl9A07MrqTI.jpg","overview":"A young girl, Chihiro, becomes trapped in a strange new world of spirits. When her parents undergo a mysterious transformation, she must call upon the courage she never knew she had to free her family.","budget":15000000,"revenue":274925095,"genres":["Animation","Family","Fantasy"],"runtime":125},{"id":240,"title":"The Godfather: Part II","tagline":"I don't feel I have to wipe everybody out, Tom. Just my enemies.","vote_average":8.4,"vote_count":4308,"release_date":"1974-12-20","poster_path":"https://image.tmdb.org/t/p/w500/bVq65huQ8vHDd1a4Z37QtuyEvpA.jpg","overview":"In the continuing saga of the Corleone crime family, a young Vito Corleone grows up in Sicily and in 1910s New York. In the 1950s, Michael Corleone attempts to expand the family business into Las Vegas, Hollywood and Cuba.","budget":13000000,"revenue":102600000,"genres":["Drama","Crime"],"runtime":200},{"id":449176,"title":"Love, Simon","tagline":"He's done keeping his story straight.","vote_average":8.4,"vote_count":82,"release_date":"2018-03-16","poster_path":"https://image.tmdb.org/t/p/w500/5YUYg5q7QfC4IoNwNUtiwdiYKPr.jpg","overview":"Everyone deserves a great love story. But for seventeen-year old Simon Spier it’s a little more complicated: he’s yet to tell his family or friends he’s gay and he doesn’t actually know the identity of the anonymous classmate he’s fallen for online.","budget":17000000,"revenue":11800000,"genres":["Comedy","Drama","Romance"],"runtime":110},{"id":424,"title":"Schindler's List","tagline":"Whoever saves one life, saves the world entire.","vote_average":8.4,"vote_count":5574,"release_date":"1993-11-29","poster_path":"https://image.tmdb.org/t/p/w500/yPisjyLweCl1tbgwgtzBCNCBle.jpg","overview":"The true story of how businessman Oskar Schindler saved over a thousand Jewish lives from the Nazis while they worked as slaves in his factory during World War II.","budget":22000000,"revenue":321365567,"genres":["Drama","History","War"],"runtime":195},{"id":122,"title":"The Lord of the Rings: The Return of the King","tagline":"The eye of the enemy is moving.","vote_average":8.3,"vote_count":9857,"release_date":"2003-12-01","poster_path":"https://image.tmdb.org/t/p/w500/rCzpDGLbOoPwLjy3OAm5NUPOTrC.jpg","overview":"Aragorn is revealed as the heir to the ancient kings as he, Gandalf and the other members of the broken fellowship struggle to save Gondor from Sauron's forces. Meanwhile, Frodo and Sam bring the ring closer to the heart of Mordor, the dark lord's realm.","budget":94000000,"revenue":1118888979,"genres":["Adventure","Fantasy","Action"],"runtime":201},{"id":550,"title":"Fight Club","tagline":"Mischief. Mayhem. Soap.","vote_average":8.3,"vote_count":11996,"release_date":"1999-10-15","poster_path":"https://image.tmdb.org/t/p/w500/adw6Lq9FiC9zjYEpOqfq03ituwp.jpg","overview":"A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy. Their concept catches on, with underground \"fight clubs\" forming in every town, until an eccentric gets in the way and ignites an out-of-control spiral toward oblivion.","budget":63000000,"revenue":100853753,"genres":["Drama"],"runtime":139},{"id":510,"title":"One Flew Over the Cuckoo's Nest","tagline":"If he's crazy, what does that make you?","vote_average":8.3,"vote_count":3872,"release_date":"1975-11-18","poster_path":"https://image.tmdb.org/t/p/w500/2Sns5oMb356JNdBHgBETjIpRYy9.jpg","overview":"While serving time for insanity at a state mental hospital, implacable rabble-rouser, Randle Patrick McMurphy inspires his fellow patients to rebel against the authoritarian rule of head nurse, Mildred Ratched.","budget":3000000,"revenue":108981275,"genres":["Drama"],"runtime":133}];
