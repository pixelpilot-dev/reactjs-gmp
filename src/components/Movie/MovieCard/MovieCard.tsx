/* eslint-disable no-param-reassign */
import React, { useState } from 'react';
import cn from 'classnames';
import { QUERY_PARAMS_BY_MOVIES, NOTIFICATION_TYPES } from '../../../core/constants';
import { I18Y, LOCALE } from '../../../core/i18y';
import { IMovieCardProps } from './interfaces';
import { TNotificationType } from '../../../core/types/TNotificationType';
import { getYear } from '../../../core/utils/date';
import { replaceNotFoundImage } from '../../../core/utils/replaceNotFoundImage';
import useQueryParam from '../../../hooks/useQueryParam';

import { MovieContextMenu } from '../MovieContextMenu';
import { Modal } from '../../UI/Modal';
import { EditMovie } from '../EditMovie';
import { DeleteMovie } from '../DeleteMovie';

import styles from './MovieCard.module.scss';

export const MovieCard: React.FC<IMovieCardProps> = ({ movie, className, onSuccessModal }) => {
  const [, setMovieQuery] = useQueryParam(QUERY_PARAMS_BY_MOVIES.MOVIE);
  const [movieEditing, setMovieEditing] = useState(null);
  const [movieDeleting, setMovieDeleting] = useState(null);
  const [isOpenEditModal, setStateEditModal] = useState(false);
  const [isOpenDeleteModal, setStateDeleteModal] = useState(false);

  const onEdit = (id: string | number) => {
    setStateEditModal(true);
    setMovieEditing(id);
  };

  const onDelete = (id: string | number) => {
    setStateDeleteModal(true);
    setMovieDeleting(id);
  };

  const onEditSuccess = (state: boolean) => {
    if (!state) {
      return;
    }

    const notification = {
      title: I18Y[LOCALE].NOTIFICATION_TITLE_SUCCESS,
      message: I18Y[LOCALE].NOTIFICATION_MESSAGE_SUCCESS_FOR_EDIT_MOVIE,
      type: NOTIFICATION_TYPES.SUCCESS as TNotificationType,
    };
    setStateEditModal(false);
    onSuccessModal(notification);
  };

  const onDeleteSuccess = (state: boolean) => {
    if (!state) {
      return;
    }

    const notification = {
      title: I18Y[LOCALE].NOTIFICATION_TITLE_SUCCESS,
      message: I18Y[LOCALE].NOTIFICATION_MESSAGE_SUCCESS_FOR_DELETE_MOVIE,
      type: NOTIFICATION_TYPES.SUCCESS as TNotificationType,
    };
    setStateDeleteModal(false);
    onSuccessModal(notification);
  };

  const handlerViewDetails = (
    event: React.MouseEvent<HTMLDivElement>,
    idMovie: string | number,
  ) => {
    event.preventDefault();

    setMovieQuery(idMovie);
    window.scrollTo({ top: 0, behavior: 'smooth' });
  };

  const renderItem = (item: string) => {
    return (
      <span data-testid='genre' key={item} className={styles.genre}>
        {item}
      </span>
    );
  };

  const renderList = (items: string[]) => items.map(renderItem);

  const modalEdit = isOpenEditModal ? (
    <Modal isOpen={isOpenEditModal} onClose={setStateEditModal}>
      <EditMovie id={movieEditing} onSuccess={onEditSuccess} />
    </Modal>
  ) : null;

  const modalDelete = isOpenDeleteModal ? (
    <Modal isOpen={isOpenDeleteModal} onClose={setStateDeleteModal}>
      <DeleteMovie id={movieDeleting} onSuccess={onDeleteSuccess} />
    </Modal>
  ) : null;

  const { id, title, poster_path, genres, release_date } = movie;

  return (
    <div data-testid='movieCard' className={cn(styles.card, className)}>
      <MovieContextMenu className={styles.menuContext}>
        <button type='button' className={styles.actionBtn} onClick={() => onEdit(id)}>
          {I18Y[LOCALE].BUTTON_EDIT}
        </button>
        <button type='button' className={styles.actionBtn} onClick={() => onDelete(id)}>
          {I18Y[LOCALE].BUTTON_DELETE}
        </button>
      </MovieContextMenu>

      <div
        data-testid='showDetailsOnPicture'
        role='button'
        aria-hidden='true'
        className={styles.preview}
        onClick={event => handlerViewDetails(event, id)}
      >
        <img
          src={poster_path}
          alt={title}
          onError={event => replaceNotFoundImage(event, styles.notFound)}
        />
      </div>

      <div className={styles.info}>
        <div
          data-testid='showDetailsOnTitle'
          role='button'
          aria-hidden='true'
          className={styles.title}
          onClick={event => handlerViewDetails(event, id)}
        >
          {title}
        </div>
        <span className={styles.year}>{getYear(release_date)}</span>
      </div>
      <div className={styles.genres}>{renderList(genres)}</div>

      {modalEdit}
      {modalDelete}
    </div>
  );
};
