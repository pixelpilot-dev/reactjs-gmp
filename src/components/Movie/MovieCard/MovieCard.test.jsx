import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MOVIE } from '../../../core/mocks';

import { MovieCard } from './MovieCard';

const spyScrollTo = jest.fn();
global.scrollTo = spyScrollTo;

jest.mock('../EditMovie', () => ({
  __esModule: true,
  EditMovie: () => {
    return <div>Edit movie</div>;
  },
}));

jest.mock('../DeleteMovie', () => ({
  __esModule: true,
  DeleteMovie: () => {
    return <div>Delete movie</div>;
  },
}));

const mockedOnSuccessModal = jest.fn();
const mockMovie = MOVIE;

describe('MovieCard component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MovieCard movie={mockMovie} onSuccessModal={mockedOnSuccessModal} />,
      { wrapper: BrowserRouter }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render all genres', () => {
    render(
      <MovieCard movie={mockMovie} onSuccessModal={mockedOnSuccessModal} />,
      { wrapper: BrowserRouter }
    );

    const genres = screen.getAllByTestId('genre');

    expect(genres.length).toBe(mockMovie.genres.length);
  });

  it('show details on click picture', async () => {
    const user = userEvent.setup();

    render(
      <MovieCard movie={mockMovie} onSuccessModal={mockedOnSuccessModal} />,
      { wrapper: BrowserRouter }
    );

    await user.click(screen.getByTestId('showDetailsOnPicture'));

    expect(spyScrollTo).toHaveBeenCalledWith({
      top: 0,
      behavior: 'smooth',
    });
  });

  it('show details on click title', async () => {
    const user = userEvent.setup();

    render(
      <MovieCard movie={mockMovie} onSuccessModal={mockedOnSuccessModal} />,
      { wrapper: BrowserRouter }
    );

    await user.click(screen.getByTestId('showDetailsOnTitle'));

    expect(spyScrollTo).toHaveBeenCalledWith({
      top: 0,
      behavior: 'smooth',
    });
  });

  it('click edit button', async () => {
    const user = userEvent.setup();

    render(
      <MovieCard movie={mockMovie} onSuccessModal={mockedOnSuccessModal} />,
      { wrapper: BrowserRouter }
    );

    const [btnOpen] = screen.getAllByRole('button');
    await user.click(btnOpen);

    const btnEdit = screen.getByRole('button', { name: 'Edit' });

    await user.click(btnEdit);
    expect(screen.getByText('Edit movie')).toBeInTheDocument();
  });

  it('click delete button', async () => {
    const user = userEvent.setup();

    render(
      <MovieCard movie={mockMovie} onSuccessModal={mockedOnSuccessModal} />,
      { wrapper: BrowserRouter }
    );

    const [btnOpen] = screen.getAllByRole('button');
    await user.click(btnOpen);

    const btnDelete = screen.getByRole('button', { name: 'Delete' });

    await user.click(btnDelete);

    expect(screen.getByText('Delete movie')).toBeInTheDocument();
  });
});
