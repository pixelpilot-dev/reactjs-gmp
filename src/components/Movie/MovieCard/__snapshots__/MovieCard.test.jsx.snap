// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="card"
    data-testid="movieCard"
  >
    <div
      class="menu menuContext"
      data-testid="menu"
    >
      <button
        class="btnOpen"
        type="button"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="#undefined"
          />
        </svg>
      </button>
      <button
        class="btnClose"
        type="button"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="#undefined"
          />
        </svg>
      </button>
    </div>
    <div
      aria-hidden="true"
      class="preview"
      data-testid="showDetailsOnPicture"
      role="button"
    >
      <img
        alt="Wonder"
        src="https://image.tmdb.org/t/p/w500/ouYgAatYH4JzIThj6FI3UYf31RI.jpg"
      />
    </div>
    <div
      class="info"
    >
      <div
        aria-hidden="true"
        class="title"
        data-testid="showDetailsOnTitle"
        role="button"
      >
        Wonder
      </div>
      <span
        class="year"
      >
        2017
      </span>
    </div>
    <div
      class="genres"
    >
      <span
        class="genre"
        data-testid="genre"
      >
        Drama
      </span>
      <span
        class="genre"
        data-testid="genre"
      >
        Family
      </span>
    </div>
  </div>
</DocumentFragment>
`;
